bukkit/rpk-unconsciousness-bukkit/src/main/kotlin/com/rpkit/unconsciousness/bukkit/unconsciousness/RPKUnconsciousnessServiceImpl.kt

```diff
@@ -24,31 +24,33 @@ import org.bukkit.potion.PotionEffect
 import org.bukkit.potion.PotionEffectType
 import java.time.LocalDateTime
 import java.time.temporal.ChronoUnit
+import java.util.concurrent.CompletableFuture
 
 
 class RPKUnconsciousnessServiceImpl(override val plugin: RPKUnconsciousnessBukkit) : RPKUnconsciousnessService {
 
-    override fun isUnconscious(character: RPKCharacter): Boolean {
+    override fun isUnconscious(character: RPKCharacter): CompletableFuture<Boolean> {
         val unconsciousStateTable = plugin.database.getTable(RPKUnconsciousStateTable::class.java)
-        val unconsciousState = unconsciousStateTable.get(character)
-        return if (unconsciousState == null) {
-            false
-        } else {
-            unconsciousState.deathTime.plus(plugin.config.getLong("unconscious-time"), ChronoUnit.MILLIS) > LocalDateTime.now()
+        return unconsciousStateTable[character].thenApply { unconsciousState ->
+            return@thenApply if (unconsciousState == null) {
+                false
+            } else {
+                unconsciousState.deathTime.plus(plugin.config.getLong("unconscious-time"), ChronoUnit.MILLIS) > LocalDateTime.now()
+            }
```

Comment:
```suggestion
            return@thenApply unconsciousState != null && unconsciousState.deathTime.plus(plugin.config.getLong("unconscious-time"), ChronoUnit.MILLIS) > LocalDateTime.now()
```
(This is minor, feel free to ignore)